### ISSUE TITLE
Consider supporting save/restore of individual containers within/from a session

### ISSUE THREAD

**Issue by ddurst** [mozilla/multi-account-containers#1325]
Referencing https://bugzilla.mozilla.org/show_bug.cgi?id=1512802, I'm filing this here to reflect a desired capability regarding containers, which has a dependency on/interaction with session restore.

If I understand it correctly, the desire is to be able to save and restore individual containers in a session (and from a session), rather than restoring everything. I moved this out of WebExtensions because it's an enhancement that runs deeper than that. Feel free to triage as necessary.

**Comment by groovecoder** [comment]
To make sure I understand if/how this would work in MAC ...

When a user views a Container's detail panel in the pop-up ...

![screen shot 2019-01-11 at 11 08 48 am](https://user-images.githubusercontent.com/71928/51048706-50b52200-1591-11e9-8a5d-3907ac5b6fc6.png)

they see a new item (next to the hide & move options) that says "Save Container tabs".

Later, they can view a Container's detail panel in the pop-up and choose "Restore Container tabs" ?

How would this be different from the "Show/Hide" feature we already have?

**Comment by ddurst** [comment]
My assumption (and I apologize, as I'm not the original reporter) is that that's where the desire is to integrate it with session restore. There's an ongoing conversation about some users wanting more control over session save/restore -- and my read of this is that (some, or these reporting) users would like to be able to choose which containers to restore from a saved session (a saved session which includes multiple containers, whether or not they were saved themselves).

**Comment by lpwarner** [comment]
@groovecoder So the status quo Show/Hide is currently the equivalent to having an add-on like [Session Manager](https://addons.mozilla.org/en-US/firefox/addon/tab-session-manager/?src=search)? Does it return the session to the exact same state?

**Comment by groovecoder** [comment]
@lpwarner No, I wouldn't say the exact same state. Hide simply closes the tabs and stores the urls of them. Show opens new tabs with the same URLs. So there's plenty of state that could change between on either client or server sides - e.g., Firefox user could clear their cookies, or serve could expire sessions, etc.

**Comment by groovecoder** [comment]
@ddurst Ah, "which containers to restore from a saved session" does seem like an enhancement request.

The `sessions` API `Session` object already returns `tabs.Tab` objects which contain the `cookieStoreId` value that maps to Containers, but I can't quite tell how all those might fit together to make this possible using just an extension.

**Comment by Nehemoth** [comment]
I would like to see this. Would be very useful.

**Comment by lpwarner** [comment]
Thanks!

**Comment by JohnLGalt** [comment]
Have to agree, Being able to, at thte very least, actually save (and later restore) individual containers as a group in the session store would be great.

An even better option would be to be able to pick and choose which containers (including all or none) are saved in / restored from the session store.

**Comment by RobertAb1** [comment]
SESSION MANAGEMENT AND CONTAINERS:

(1)
Consider assigning different sessions to separate containers
https://bugzilla.mozilla.org/show_bug.cgi?id=1524959
https://bugzilla.mozilla.org/show_bug.cgi?id=1528598#c6
https://bugzilla.mozilla.org/show_bug.cgi?id=1528604#c2

(2)
Consider allowing individual session management of containers
https://bugzilla.mozilla.org/show_bug.cgi?id=1512802

Consider supporting save/restore of individual containers within/from a session (GitHub)
https://github.com/mozilla/multi-account-containers/issues/1325
